homeassistant/components/osoenergy/binary_sensor.py

```diff
@@ -0,0 +1,95 @@
+"""Support for OSO Energy binary sensors."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from apyosoenergyapi import OSOEnergy
+from apyosoenergyapi.helper.const import OSOEnergyBinarySensorData
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import OSOEnergyEntity
+from .const import DOMAIN
+
+
+@dataclass(frozen=True, kw_only=True)
+class OSOEnergyBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Class describing OSO Energy heater binary sensor entities."""
+
+    value_fn: Callable[[OSOEnergy], bool]
+
+
+SENSOR_TYPES: dict[str, OSOEnergyBinarySensorEntityDescription] = {
+    "power_save": OSOEnergyBinarySensorEntityDescription(
+        key="power_save",
+        translation_key="power_save",
+        value_fn=lambda entity_data: entity_data.state,
+    ),
+    "extra_energy": OSOEnergyBinarySensorEntityDescription(
+        key="extra_energy",
+        translation_key="extra_energy",
+        value_fn=lambda entity_data: entity_data.state,
+    ),
+    "heater_state": OSOEnergyBinarySensorEntityDescription(
+        key="heater_state",
+        translation_key="heater_state",
+        value_fn=lambda entity_data: entity_data.state,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up OSO Energy binary sensor."""
+    osoenergy = hass.data[DOMAIN][entry.entry_id]
```

Comment:
I think adding the type hint here will help
```suggestion
    osoenergy:OSOEnergy = hass.data[DOMAIN][entry.entry_id]
```